offline/coturn.md

```diff
@@ -0,0 +1,471 @@
+
+## Configure the Coturn labels to select on which machine(s) it will run.
+
+Next, we must select on which machine Coturn will run.
+
+In this example, we've decided it will run on the third kubernetes node, `kubenode3`, which has an IP address of `192.168.122.23`.
+
+We've set the `nodeSelector` in the `values.yaml` file to select the `coturn` role, this and machine we label with the `wire.com/role: coturn` label will be selected to run Coturn.
+
+So we need to label the `kubenode3` machine with the `wire.com/role: coturn` label.
+
+We do this by running:
+
+```bash
+
+d kubectl label node kubenode3 wire.com/role=coturn
+
+```
+
+By default, only one machine will be selected to run Coturn. 
+
+If you want to run Coturn on multiple machines, you must:
+
+1. Add the `wire.com/role: coturn` label to multiple machines.
+
+2. Change the `replicaCount` in the `charts/coturn/values.yaml` file to the number of machines you want to run Coturn on.
+
+## Configure the SFT labels for Coturn and SFT to share a port range. 
+
+```{note}
+
+Note this step is also documented in the Switch to [ Wire install docs](docs_ubuntu_22.04.md)
+
+``` 
+
+We must make sure that Coturn pods and SFT pods do not run on the same kubernetes nodes.
+
+This means we must label the kubernetes nodes to run on nodes that we did not select to run Coturn in the previous step.
+
+In this example, we've decided to run Coturn on the first kubernetes node, `kubenode1`, which has an IP address of `192.168.122.21`.
+
+First we make sure the SFT chart is configured to only run on kubernetes nodes with the right label (`sftd`).
+
+Edit the `values/sftd/values.yaml` file:
+
+```bash
+
+nodeSelector:
+  wire.com/role: sftd
+
+``` 
+
+Then we label the `kubenode1` machine with the `wire.com/role: coturn` label:
+
+```bash
+
+d kubectl label node kubenode1 wire.com/role=sftd
+
+```
+
+If we want to run SFT on multiple nodes, the procedure is the same as the one documented above for running Coturn on multiple nodes.
+
+We now should have Coturn configured to run on one or more kubernetes node(s), and SFT configured to run on one or more kubernetes node(s), and the two should not run on the same node(s)/overlap.
+
+Before moving on, we must also re-deploy SFT's chart to apply the new configuration:
+
+```bash
+
+d helm upgrade --install sftd ./charts/sftd --set 'nodeSelector.wire\.com/role=sftd' --values values/sftd/values.yaml
+
+```
+
+## Configure the port redirection in Nftables.
+
+We must configure the port redirection in Nftables to allow traffic to reach Coturn and SFT.
+
+Calling and TURN services (Coturn, Restund, SFT) require being reachable on a range of ports used to transmit the calling data.
+
+Because SFT and Coturn can not run on the same node, and can not share the same port range on the "outside" interface of the machine, we must configure the port redirection in Nftables to allow traffic to reach Coturn and SFT by splitting the ports between the two services.
```

Comment:
I am not certain this is true. it's MORE true that they both want to use the same port range, so predicting which nodes are using what ports ahead of time requires dividing ahead of time.